GitExtUtils/ArgumentString.cs

```diff
@@ -1,24 +1,23 @@
-﻿using System;
-using System.Diagnostics;
-using GitCommands;
+﻿using System.Diagnostics;
 using JetBrains.Annotations;
 
-namespace GitUIPluginInterfaces
+namespace GitCommands
 {
     [DebuggerDisplay("{" + nameof(Arguments) + "}")]
     public readonly struct ArgumentString
     {
         [CanBeNull] public string Arguments { get; }
+        public int? Length { get => Arguments?.Length; }
 
-        private ArgumentString([NotNull] string arguments)
+        private ArgumentString([CanBeNull] string arguments)
```

Comment:
Why do you allow null here?

Reply:
Because the tostring will turn it to a "". I'm thinking of changing the scope of Arguments too. Really should only have one way to get the string back.

Reply:
It is not valid to pass null here. I would not change the attribute or validation. Note that because this is a struct you _can_ end up with a null field internally, but that's only possible if using `default(ArgumentString)` or similar which doesn't use the ctor. We should not be changing this code path as it could hide errors.

Reply:
I did this to fix an issue with VS fussing about the conversion in the optional parameter.  It looks like VS was yelling for nothing.  I'll revert this.